charts/linkerd2/templates/config.yaml

```diff
@@ -34,3 +34,24 @@ data:
   {{- else -}}
   {{- include "linkerd.configs.install" . | nindent 4}}
   {{- end }}
+  values: |
+  {{- $values := deepCopy .Values }}
+  {{- /*
+    WARNING! All sensitive or private data such as TLS keys must be removed
+    here to avoid it being publicly readable.
+  */ -}}
+  {{- if kindIs "map" $values.identity.issuer.tls -}}
+    {{- $_ := unset $values.identity.issuer.tls "keyPEM"}}
+  {{- end -}}
+  {{- if kindIs "map" $values.profileValidator -}}
+    {{- $_ := unset $values.profileValidator "keyPEM"}}
+  {{- end -}}
+  {{- if kindIs "map" $values.proxyInjector -}}
+    {{- $_ := unset $values.proxyInjector "keyPEM"}}
+  {{- end -}}
+  {{- if kindIs "map" $values.tap -}}
+    {{- $_ := unset $values.tap "keyPEM"}}
```

Comment:
Why the additional `kindIs` checks? It doesn't fail to me if they're absent.

Reply:
I hit certain tests which failed without it.  I believe when `Values.identity.issuer.tls` is absent (or nil, maybe?).  Here is the failure:

https://github.com/linkerd/linkerd2/runs/1184792150?check_suite_focus=true